lib/linalg/BasisWriter.cpp

```diff
@@ -53,96 +52,77 @@ BasisWriter::BasisWriter(
     char tmp2[100];
     sprintf(tmp2, "_snapshot.%06d", rank);
     snap_file_name = base_file_name + tmp2;
+
+    // create and open snapshot/basis database
+    CAROM_VERIFY(db_format_ == Database::HDF5);
+    d_snap_database = new HDFDatabase();
+    d_database = new HDFDatabase();
 }
 
 BasisWriter::~BasisWriter()
 {
-    if (d_database) {
-        d_database->putInteger("num_time_intervals", d_num_intervals_written);
-        d_database->close();
-        delete d_database;
-    }
-    if (d_snap_database) {
-        d_snap_database->putInteger("num_time_intervals", d_num_intervals_written);
-        d_snap_database->close();
-        delete d_snap_database;
-    }
+    delete d_database;
+    delete d_snap_database;
 }
 
 void
 BasisWriter::writeBasis(const std::string& kind)
 {
-
     CAROM_ASSERT(kind == "basis" || kind == "snapshot");
 
     char tmp[100];
-    double time_interval_start_time =
-        d_basis_generator->getBasisIntervalStartTime(d_num_intervals_written);
-    sprintf(tmp, "time_%06d", d_num_intervals_written);
+    // This 0 index is the remainder from time interval concept.
```

Comment:
Why can't we remove the 0 index now that we decided not to preserve backward compatibility for time intervals?

Reply:
Backward compatibility can be kept/removed at various level. We decide to break the code compatibility, so all the softwares depending on libROM should update their codes in order to build/compile. But I intended to keep data compatibility, so that all the previous data can be used seamlessly without additional conversion process.

I guess we can come up with a python script for dataset name conversion. Would that be a better path to you?

Reply:
I'm not sure it is common practice to store data long term, although in theory someone may be doing that. I suppose most users generate FOM data, build ROMs, and test, without saving data for use in the future. In the rare case that someone is saving a lot of data, your suggestion to rename with a script should suffice, and we can leave that to the users. I would prefer that to maintaining indices in the code that no longer make sense.

Reply:
> I'm not sure it is common practice to store data long term, although in theory someone may be doing that. 

At least I'm the one keeping the snapshot and basis data. But I'm okay with having a format converter, so open to either route.